test/type_traits/compatible_sentinal.cpp

```diff
@@ -0,0 +1,29 @@
+#include <immer/detail/type_traits.hpp>
```

Comment:
The filename has a typo `sentinal` -> `sentinel`

Reply:
You have created lost of test files. I am very happy that you are taking testing seriously! However, I do not think it is necessary to have a different test-file for every different trait. If you ask me, I'd rather merge all these tests in a single file in `test/detail/type_traits.cpp` (also, you did not put the detail in the path of the test files, but for it should be there so it is clearer that these are tests about the internal implementation not the public API.)

Reply:
Will do!